core/tee/tadb.c

```diff
@@ -172,13 +172,13 @@ static TEE_Result set_file(struct tee_tadb_dir *db, int idx)
 
 static void clear_file(struct tee_tadb_dir *db, int idx)
 {
-	if (idx < db->nbits)
+	if (db->files && idx < db->nbits)
```

Comment:
If `db->nbits` is larger than `0` then will `db->files` always be non-NULL, see `maybe_grow_files()`, so this test is meaningless. We can add a comment since it may not be obvious. Same for `test_file()` below.

Reply:
I'll swap that for a comment. Thank you, these dependencies are not always obvious. Maybe some general documentation for `struct tee_tadb_dir` would be even better.